dependencies/gtest/CMakeLists.txt

```diff
@@ -26,6 +26,10 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "SunPro")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64")
 endif()
 
+if(WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m64 /D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING")
```

Comment:
What is giving this warning in our sources?

Reply:
It's in gtest: gtest-port.h

Reply:
This isn't how you do `/D` or `-D` flags in CMake. Also it should be defined in the gtest library so that anything compiling/linking with gtest gets this flag.

Reply:
See `boost/CMakeList` for
```
target_compile_definitions(${PROJECT_NAME} INTERFACE
  BOOST_ALL_NO_LIB
)
```
and add it there.
```
target_compile_definitions(${PROJECT_NAME} INTERFACE
  BOOST_ALL_NO_LIB
  _SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING
)
```
Even better would be to wrap around `if (MSVC)` conditional.